### PR TITLE
added snake game

### PR DESCRIPTION
**Pull request by ShrishtiSingh26**
Closes: Issue #112 

- **Title**:  adding a snake game under games section #112
- **Your Name**: Shrishti Singh
- **Idenitfy yourself**:  Contributor in gssoc-ext and Hacktoberfest

### Describe the add-ons or changes you've made 📃

Give a clear description of what have you added or modifications made
Key Points:
Game State Management: All the game state (snake, food, direction, etc.) is stored in st.session_state to persist across user actions.
Grid Display: The game grid is displayed using emojis (🟩 for the snake and 🍎 for the food), but you can customize this.
Movement Control: The player controls the snake with W, A, S, D keys via Streamlit buttons.
Game Logic: The snake moves, eats food, grows, and checks for collisions with walls or itself.
Restart Option: A restart button re-initializes the game state.

### Checklist: ☑️

- [x] My code follows the [Contributing Guidelines](https://github.com/Avdhesh-Varshney/Jarvis/blob/main/Contributing.md) & [Code of Conduct](https://github.com/Avdhesh-Varshney/Jarvis/blob/main/CODE_OF_CONDUCT.md) of this project.
- [x] This PR does not contain plagiarized content.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly wherever it was hard to understand.
- [x] My changes generate no new warnings.

### Screenshots 📷

image
image

### CODE REVIEW

**Review by Avdhesh-Varshney — CHANGES_REQUESTED** [review]
Never create your pr by committing in `main` branch.

File: src/apps/pages/programs/Games/snakegame.py [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+        st.error(f"Game Over! Your Score: {st.session_state.score}")
+    else:
+        st.success(f"Score: {st.session_state.score}")
+
+    # Restart the game
+    if st.button("Restart Game"):
+        initialize_game()
+
+# Run the game
+snake_game()
```
[the remark this answers is not in the document]
Comment (ShrishtiSingh26) [reply src/apps/pages/programs/Games/snakegame.py:109]: @Avdhesh-Varshney  please elaborate on this issue..

File: src/apps/pages/programs/Games/snakegame.py [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+        st.session_state.direction = "UP"
+    elif key == "s" and st.session_state.direction != "UP":
+        st.session_state.direction = "DOWN"
+    elif key == "a" and st.session_state.direction != "RIGHT":
+        st.session_state.direction = "LEFT"
+    elif key == "d" and st.session_state.direction != "LEFT":
+        st.session_state.direction = "RIGHT"
+
+# Main Snake Game function
+def snake_Game():
```
Comment (Avdhesh-Varshney) [inline src/apps/pages/programs/Games/snakegame.py:79]: - Remove underscore.
- Review other python programs.

**Review by Avdhesh-Varshney — CHANGES_REQUESTED** [review]
Attach a working video.

**Comment by ShrishtiSingh26** [comment]
@Avdhesh-Varshney resolved the raised issues in PR #117

**Pull request closed without merging**